core/src/main/java/org/infinispan/marshall/core/Primitives.java

```diff
@@ -177,13 +179,13 @@ private static void writeByteArray(byte[] obj, BytesObjectOutput out) {
       final int len = obj.length;
       if (len == 0) {
          out.writeByte(ID_ARRAY_EMPTY);
-      } else if (len <= ID_SMALL) {
+      } else if (len <= SMALL_ARRAY_MAX) {
          out.writeByte(ID_ARRAY_SMALL);
-         out.writeByte(len - 1);
+         out.writeByte(len - SMALL_ARRAY_MIN);
          out.write(obj, 0, len);
-      } else if (len <= ID_MEDIUM) {
+      } else if (len <= MEDIUM_ARRAY_MAX) {
          out.writeByte(ID_ARRAY_MEDIUM);
-         out.writeShort(len - 1);
+         out.writeShort(len - MEDIUM_ARRAY_MIN);
```

Comment:
Why change this? Is there a real difference between writing a smaller or larger short? This is what `writeShort` does:

```
      int newcount = ensureCapacity(2);
      final int s = pos;
      bytes[s] = (byte) (v >> 8);
      bytes[s+1] = (byte) v;
      pos = newcount;
```